diff --git a/toolkit/library/Makefile.in b/toolkit/library/Makefile.in
index 1101c2f..52d337e 100644
--- a/toolkit/library/Makefile.in
+++ b/toolkit/library/Makefile.in
@@ -558,6 +558,10 @@ ifdef MOZ_GSTREAMER
 EXTRA_DSO_LDOPTS += $(GSTREAMER_LIBS)
 endif
 
+ifdef MOZ_CCNX
+EXTRA_DSO_LDOPTS += $(CCNX_LIBS)
+endif
+
 include $(topsrcdir)/config/rules.mk
 
 export:: $(RDF_UTIL_SRC_CPPSRCS) $(INTL_UNICHARUTIL_UTIL_CPPSRCS)
