diff --git a/modules/libpref/src/init/all.js b/modules/libpref/src/init/all.js
index 06ca000..8fb3e04 100644
--- a/modules/libpref/src/init/all.js
+++ b/modules/libpref/src/init/all.js
@@ -199,7 +199,9 @@ pref("media.webm.enabled", true);
 #ifdef MOZ_GSTREAMER
 pref("media.h264.enabled", true);
 #endif
-
+#ifdef MOZ_GSTREAMER
+pref("media.ccnx.enabled", true);
+#endif
 
 // Whether to autostart a media element with an |autoplay| attribute
 pref("media.autoplay.enabled", true);
